syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/src/org/fusesource/ide/syndesis/extensions/ui/wizards/pages/SyndesisExtensionProjectWizardExtensionDetailsPage.java

```diff
@@ -118,30 +128,52 @@ public String getExtensionDescription() {
 	}
 	
 	private void validateFields() {
-		setErrorMessage(null);
-		
+		// validate the extension id
 		if (Strings.isBlank(extensionIdText.getText())) {
-			setErrorMessage(Messages.newProjectWizardExtensionDetailsPageErrorMissingExtensionId);
+			addErrorMarkerForControl(extensionIdText, Messages.newProjectWizardExtensionDetailsPageErrorMissingExtensionId);
 			setPageComplete(false);
-			return;
 		} else if (extensionIdText.getText().indexOf(' ') != -1) {
-			setErrorMessage(Messages.newProjectWizardExtensionDetailsPageErrorInvalidExtensionId);
+			addErrorMarkerForControl(extensionIdText, Messages.newProjectWizardExtensionDetailsPageErrorInvalidExtensionId);
 			setPageComplete(false);
-			return;
+		} else {
+			cleanErrorMarkerForControl(extensionIdText);
 		}
 		
+		// validate the extension version
 		if (Strings.isBlank(extensionVersionText.getText())) {
-			setErrorMessage(Messages.newProjectWizardExtensionDetailsPageErrorMissingExtensionVersion);
+			addErrorMarkerForControl(extensionVersionText, Messages.newProjectWizardExtensionDetailsPageErrorMissingExtensionVersion);
 			setPageComplete(false);
-			return;
+		} else {
+			Pattern versionPattern = Pattern.compile("^(\\d+){1}(\\.\\d+){1}(\\.\\d+)?((\\-).*)?$");
```

Comment:
Where does this added validation comes from?
Can't we reuse some Syndesis code?

Reply:
I can't use 1.0.0.SNAPSHOT or 1.0.0.v20180105 , is it a restriction from Syndesis?

Reply:
> I can't use 1.0.0.SNAPSHOT or 1.0.0.v20180105 , is it a restriction from Syndesis?

As we are not building OSGi bundles but plain jars I would have thought its better to stick to x.y.z-something notation.

Reply:
> Where does this added validation comes from? Can't we reuse some Syndesis code?

I added it to make sure we have a valid version. Need to check the syndesis code or ask Luca to see if we can reuse something.

Reply:
With OSGi there are specifications but for plain jar I don't know which specifications exists but I think that 1.0.0.SNAPSHOT or 1.0.0.v20180105 can be used.
Have you checked if Syndesis has some specifications on the version?

Reply:
> Need to check the syndesis code or ask Luca to see if we can reuse something.

Reply:
I am also fine with allowing both - and .